pkg/controller/machine/machine.go

```diff
@@ -496,10 +502,8 @@ func (c *Controller) ensureInstanceExistsForMachine(prov cloud.Provider, machine
 			if _, errNested := c.updateMachineError(machine, machinev1alpha1.InvalidConfigurationMachineError, err.Error()); errNested != nil {
 				return fmt.Errorf("failed to update machine error after failed validation: %v", errNested)
 			}
-			c.recorder.Eventf(machine, corev1.EventTypeWarning, "ValidationFailed", "Validation failed: %v", err)
 			return fmt.Errorf("invalid provider config: %v", err)
 		}
-		c.recorder.Event(machine, corev1.EventTypeNormal, "ValidationSucceeded", "Validation succeeded")
```

Comment:
Does this mean we're now only emit the event for azure?
I dislike the idea of having to emit the event in the cloud-provider code - its redundant and easy to forget

Reply:
https://github.com/kubermatic/machine-controller/pull/288/files#diff-87acff32b00826238809db10ac9ef562R309

It  just got moved inside the wrapper around the `Validate` we have within the machine-controller

Reply:
Ah sorry - i misread my diff -  though it belongs to the azure code now

Reply:
As a future enhancement I would however propose to pass down the EventRecorder to the `Create`/`Delete` funcs, because Azure takes forever to delete and as a User it is pretty hard to distinguish if its already doing something or not